File: 03-Java-Advanced/03-Sets-and-Maps/Exercise/src/_05_Phonebook.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.Set;

public class _05_Phonebook {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();
        Map<String, String> phonebook = new HashMap<>();

        while (!input.equals("search")) {
            String[] tokens = input.split("-");
            String name = tokens[0];
            String phoneNumber = tokens[1];

            phonebook.put(name, phoneNumber);

            input = scanner.nextLine();
        }

        String command = scanner.nextLine();

        while (!command.equals("stop")) {

            if (phonebook.containsKey(command)) {
                System.out.printf("%s -> %s%n", command, phonebook.get(command));
            } else {
                System.out.printf("Contact %s does not exist.%n", command);
            }

            command = scanner.nextLine();
        }
    }
}

File: 05-JS-Front-End/02-JS-Front-End/04-Objects-And-Classes/Lab/_05_Phonebook.js
function phonebook(array) {

    let phonebook = {};

    for (let element of array) {
        let tokens = element.split(' ');

        let key = tokens[0];
        let value = tokens[1];

        phonebook[key] = value;
    }

    for (let obj in phonebook) {
        console.log(`${obj} -> ${phonebook[obj]}`); 
    }

}
